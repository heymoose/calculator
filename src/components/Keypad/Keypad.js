import React from 'react';
import Key from './Key/Key';
import keyStyle from './Key/Key.module.css';
import PropTypes from 'prop-types';
import style from './Keypad.module.css';

const Keypad = ({ callOperator, numbers, operators, setOperator, updateDisplay }) => {
    const numberKeys = numbers.map(number => (
        <Key
            key={number}
            keyAction={updateDisplay}
            keyType={keyStyle.NumberKey}
            keyValue={number}
        />
    ));

    const operatorKeys = operators.map(operator => (
        <Key
            key={operator}
            keyAction={setOperator}
            keyType={keyStyle.OperatorKey}
            keyValue={operator}
        />
    ));

    return (
        <div className={style.KeypadContainer}>
            <div className={style.NumbersContainer}>{numberKeys}</div>
            <div className={style.OperatorsContainer}>{operatorKeys}</div>
            <div className={style.SubmitContainer}>
                <Key keyAction={callOperator} keyType={keyStyle.SubmitKey} keyValue='=' />
            </div>
        </div>
    );
};

Keypad.propTypes = {
    callOperator: PropTypes.func.isRequired,
    numbers: PropTypes.array.isRequired,
    operators: PropTypes.array.isRequired,
    setOperator: PropTypes.func.isRequired,
    updateDisplay: PropTypes.func.isRequired
};

export default Keypad;
