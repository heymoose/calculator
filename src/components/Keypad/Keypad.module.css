.KeypadContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    height: var(--keypad-height);
    padding: 2%;
    width: var(--keypad-width);
}

.NumbersContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    height: 80%;
    width: 75%;
}

.OperatorsContainer {
    display: flex;
    flex-direction: column;
    height: 80%;
    width: 25%;
}

.SubmitContainer {
    height: 20%;
    width: 100%;
}
