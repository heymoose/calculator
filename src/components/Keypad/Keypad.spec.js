import React from 'react';
import { shallow } from 'enzyme';
import Keypad from './Keypad';
import keypadStyle from './Keypad.module.css';

describe('Keypad', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Keypad
                callOperator={jest.fn()}
                numbers={[]}
                operators={[]}
                setOperator={jest.fn()}
                updateDisplay={jest.fn()}
            />
        );
    });

    it("should render 3 <div />'s", () => {
        expect(wrapper.find('div').length).toEqual(3);
    });

    it('renders the values of the numbers', () => {
        wrapper.setProps({ numbers: ['0', '1', '2'] });
        expect(wrapper.find('.NumbersContainer').text()).toEqual('012');
    });

    it('renders the values of the operators', () => {
        wrapper.setProps({ operators: ['+', '-', '*', '/'] });
        expect(wrapper.find('.OperatorsContainer').text()).toEqual('+-*/');
    });
});
