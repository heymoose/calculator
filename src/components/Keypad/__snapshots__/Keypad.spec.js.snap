// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Keypad
    callOperator={[MockFunction]}
    numbers={Array []}
    operators={Array []}
    setOperator={[MockFunction]}
    updateDisplay={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="NumbersContainer"
        />,
        <div
          className="OperatorsContainer"
        />,
        <div
          className="SubmitContainer"
        >
          <key
            keyAction={[MockFunction]}
            keyType="SubmitKey"
            keyValue="="
          />
        </div>,
      ],
      "className": "KeypadContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "NumbersContainer",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "OperatorsContainer",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <key
            keyAction={[MockFunction]}
            keyType="SubmitKey"
            keyValue="="
          />,
          "className": "SubmitContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "keyAction": [MockFunction],
            "keyType": "SubmitKey",
            "keyValue": "=",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="NumbersContainer"
          />,
          <div
            className="OperatorsContainer"
          />,
          <div
            className="SubmitContainer"
          >
            <key
              keyAction={[MockFunction]}
              keyType="SubmitKey"
              keyValue="="
            />
          </div>,
        ],
        "className": "KeypadContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "NumbersContainer",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "OperatorsContainer",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <key
              keyAction={[MockFunction]}
              keyType="SubmitKey"
              keyValue="="
            />,
            "className": "SubmitContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "keyAction": [MockFunction],
              "keyType": "SubmitKey",
              "keyValue": "=",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
